packages/core/src/lib/__tests__/modal.spec.ts

```diff
@@ -1,38 +1,59 @@
 import { describe, it, expect, vi } from 'vitest';
-import { render, fireEvent } from '@testing-library/svelte';
+import { render, fireEvent, screen } from '@testing-library/svelte';
 import Modal from '../modal.svelte';
 
 describe('Modal', () => {
   it('should emit close event when close icon button is clicked', async () => {
-    const { getByTitle, component } = render(Modal, {
+    const { component } = render(Modal, {
       open: true,
     });
 
-    const closeButton = getByTitle('Close modal');
+    const closeButton = screen.getByTitle('Close modal');
     const onClose = vi.fn();
     component.$on('close', onClose);
     await fireEvent.click(closeButton);
 
-    expect(onClose).toBeCalledWith(expect.objectContaining({ detail: true }));
+    expect(onClose).toBeCalled();
   });
 
   it('should emit close event when clicked outside the modal', async () => {
-    const { container, component } = render(Modal, {
+    const { component } = render(Modal, {
       open: true,
     });
 
     const onClose = vi.fn();
     component.$on('close', onClose);
 
-    const background = container.querySelector('div[role="button"]');
-    if (!background) {
-      throw new Error('Background not found');
-    }
+    const modal = screen.getByRole('dialog');
 
-    await fireEvent.click(background);
+    await fireEvent.click(modal.parentElement!);
 
-    expect(onClose).toHaveBeenLastCalledWith(
-      expect.objectContaining({ detail: true })
-    );
+    expect(onClose).toBeCalled();
+  });
+
+  it('if open is true, modal should be visible', () => {
```

Comment:
Can we add an expectation to this test to check that `aria-modal` is set properly?